tmt/steps/report/reportportal.py

```diff
@@ -142,6 +150,15 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
              (e.g. 'Idle'). 'To Investigate' is used by default.
              """)
 
+    log_size_limit: str = field(
+        option="--log-size-limit",
+        metavar="LOG_SIZE_LIMIT",
+        default=_str_env_to_default('log_size_limit', '1048576'),
+        help="""
+             Size limit in bytes for log upload to ReportPortal.
+             The default limit is 1048576 bytes (1 MB).
```

Comment:
The default value should not be encoded in two places, in a string. Please, make it a module-level "constant", e.g. `DEFAULT_LOG_SIZE_LIMIT: int = 1048576`, then use it in calls and texts: `_str_env_to_default('log_size_limit', str(DEFAULT_LOG_SIZE_LIMIT))`, `f"""... The default limit is {DEFAULT_LOG_SIZE_LIMIT} bytes ({DEFAULT_LOG_SIZE_LIMIT / 1024 / 1024} MB)."""`

Yes, it's more work, but the magic number is now defied at one place and one place only, all three mentions are derived from this single value, and note that I defined it as an integer, because it is an integer, it's a size limit - the fact that `_str_env_to_default()` needs a string should not dictate and bend the type of this value, that's a problem of `_str_env_to_default()`.

Also, the type of this field should also be `int`, I think `normalize_int(_str_env_to_default(...))` should work, and it should always either return an integer or raise an error e.g. when a user tries `--log-size-limit=foo`.

Reply:
Yes, that makes sense and thank you for such an extensive reasoning. :-)

Reply:
Btw, for now I didn't use `normalize_int` as it requires `logger` as a 2nd argument.